src/pool-cl/CLPoolManagerOwner.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Currency} from "../types/Currency.sol";
+import {IProtocolFeeController} from "../interfaces/IProtocolFeeController.sol";
+import {PausableRole} from "../base/PausableRole.sol";
+import {ICLPoolManager} from "./interfaces/ICLPoolManager.sol";
+import {IPoolManagerOwner} from "../interfaces/IPoolManagerOwner.sol";
+
+/**
+ * @dev This contract is the owner of the CLPoolManager contract
+ *
+ * A seperate owner contract is used to handle some functionality so as to reduce the contract size
+ * of PoolManager. This allow a higher optimizer run, reducing the gas cost for other poolManager functions.
+ */
+contract CLPoolManagerOwner is IPoolManagerOwner, PausableRole {
+    ICLPoolManager public immutable poolManager;
+
+    constructor(ICLPoolManager _poolManager) {
+        poolManager = _poolManager;
+    }
+
+    /// @inheritdoc IPoolManagerOwner
+    function pausePoolManager() external override onlyPausableRoleOrOwner {
+        poolManager.pause();
+    }
+
+    /// @inheritdoc IPoolManagerOwner
+    function unpausePoolManager() external override onlyOwner {
+        poolManager.unpause();
+    }
+
+    /// @inheritdoc IPoolManagerOwner
+    function setProtocolFeeController(IProtocolFeeController protocolFeeController) external override onlyOwner {
+        poolManager.setProtocolFeeController(protocolFeeController);
+    }
+
+    /// @inheritdoc IPoolManagerOwner
+    function collectProtocolFees(address recipient, Currency currency, uint256 amount) external override onlyOwner {
+        poolManager.collectProtocolFees(recipient, currency, amount);
+    }
```

Comment:
should we remove `collectProtocolFees` and only allow protocolFeeController to collectProtocolFees ?

![CleanShot 2024-09-12 at 15 27 38](https://github.com/user-attachments/assets/bf9d465f-7699-4e9b-b649-f47f91c2df4d)


So that it's clearer only owner designated protocolFeeController can collect the fee and owner is only responsible for setProtocolFeeController

Reply:
would this be out of the scope for this PR? 

1/ if we remove owner capability, then we should have the `ProtocolFeeController` contract ready -- so maybe we should remove `owner()` in the same PR this contract is introduced 

though yeh, im fine with removing `owner` capability to `collectProtocolFees()`. what do you think @ChefSnoopy

Reply:
ok , agree

Reply:
got it. will 

1/ remove `owner()` permission from calling collectProtocolFee in PoolManager in this PR. 
2/ create an issue to track when we create `ProtocolFeeController`

Reply:
PR created on removing `msg.sender != owner()` at https://github.com/pancakeswap/pancake-v4-core/pull/174

seperate PR created as there was test cases to update and wanted to avoid polluting this PR